packages/delegation-process/src/services/validators.ts

```diff
@@ -0,0 +1,100 @@
+import {
+  DelegationKind,
+  DelegationState,
+  delegationState,
+  EServiceId,
+  PUBLIC_ADMINISTRATIONS_IDENTIFIER,
+  Tenant,
+  TenantId,
+} from "pagopa-interop-models";
+import {
+  delegationAlreadyExists,
+  delegatorAndDelegateSameIdError,
+  eserviceNotFound,
+  invalidExternalOriginError,
+  tenantNotAllowedToDelegation,
+  tenantNotFound,
+} from "../model/domain/errors.js";
+import { ReadModelService } from "./readModelService.js";
+
+/* ========= STATES ========= */
+export const delegationNotActivableStates: DelegationState[] = [
+  delegationState.rejected,
+  delegationState.revoked,
+];
+
+export const assertEserviceExists = async (
+  eserviceId: EServiceId,
+  readModelService: ReadModelService
+): Promise<void> => {
+  const eservice = await readModelService.getEServiceById(eserviceId);
+  if (!eservice) {
+    throw eserviceNotFound(eserviceId);
+  }
+};
+
+export const assertDelegatorIsNotDelegate = (
+  delegatorId: TenantId,
+  delegateId: TenantId
+): void => {
+  if (delegatorId === delegateId) {
+    throw delegatorAndDelegateSameIdError();
+  }
+};
+
+export const assertDelegatorIsIPA = async (
+  delegator?: Tenant
+): Promise<void> => {
+  if (delegator?.externalId?.origin !== PUBLIC_ADMINISTRATIONS_IDENTIFIER) {
+    throw invalidExternalOriginError(delegator?.externalId?.origin);
+  }
+};
+
+export const assertTenantAllowedToDelegation = (tenant: Tenant): void => {
```

Comment:
I would change the name of the function to make it more specific 
```suggestion
export const assertTenantAllowedToReceiveProducerDelegation = (tenant: Tenant): void => {
```

Reply:
I think there will be the  `assertTenantAllowedToReceiveConsumerDelegation` counter part

Reply:
Done in [commit](https://github.com/pagopa/interop-be-monorepo/pull/1080/commits/1f002bb15450e3640c799b5d087b001534de98b9)